Add experimental warning (#1)
single-spa/standalone-single-spa-webpack-plugin: README.md

A webpack plugin for running single-spa microfrontends in standalone mode. This is an alternative to using [import-map-overrides](https://github.com/joeldenning/import-map-overrides).

> **⚠️ Warning: Experimental**
>

> > This plugin should be considered "experimental" in that the core team has not yet settled on a consensus as to if 'standalone mode' is how local development should be approached. Generally speaking, developers want this because "that's the way we used to do it". **There are potential pitfalls that you should understand before deciding to use this plugin.**

Reviewer: > developers want this because "that's the way we used to do it"

This is a bit pointed / sharp, and sort of blames the developer for wanting something, which I think is a bit too strong. Maybe this?

```suggestion
> This plugin should be considered "experimental" since it is new and is not equivalent to developing in integrated mode. The single-spa core team created this plugin because it's an often-requested feature, but we're not sure if it will result in a better experience for our users. **There are potential pitfalls that you should understand before deciding to use this plugin.**
```

Author: I don't see how it blames developers but rather explains the source of the mentality of the request. My experience is that the devs asking for this are beginners to single-spa who want to hold on to old paradigms. But I'll try to reword it.  Does this reach a middle point?

```suggestion
> This plugin should be considered "experimental" since it is new. The single-spa core team created this plugin because it's an often-requested feature, usually as a hold-over from old paradigms. But we're not sure if it will result in a better experience for developers. **There are potential pitfalls that you should understand before deciding to use this plugin.**
```

Reviewer: > as a hold-over from old paradigms

I would contend that running in standalone mode is not an old paradigm. One of the things that convinced me that this standalone plugin is needed is that module federation fully encourages standalone mode as the default development mechanism, and module federation will perhaps soon overtake single-spa as the dominant microfrontends solution.

In other words, I don't think that running a web app locally as a standalone app is an old paradigm, since the new paradigm (module federation) encourages it.

Reviewer: > I don't see how it blames developers

The quotes around `"that's the way we used to do it"` come off as air quotes that are perhaps condescending.

>
> - 'standalone' also means isolated which may lead to situations of "works on my machine, but not on the deployed environment."

> > - The import map used in this plugin's configuration may quickly become outdated if you use any sort of versioning (which is recommended as a best-practice in general). This may lead to developing locally against a package whose API has changed.

Reviewer: ```suggestion
> - This plugin automatically upgrades to the latest versions of SystemJS and single-spa, which is likely different than most environments, which pin to a specific version.
```

Author: These seem to be different and separate points to me. Mine is talking about how a hard-coded import map in this plugin's config can quickly become outdated. Yours talks about automatic upgrades to SystemJS and single-spa script tags. I think should both be included.

Reviewer: I see, yes that's true. Regarding hard coding the import map - the better solution is to provide the `importMapUrl`, which allows you to always use the latest deployed version of your other microfrontends. See for instance this code:

https://github.com/react-microfrontends/people/blob/25f18f22ee89fea827db02cb92d3e021ea514938/webpack.config.js#L22-L24

If you do it this way, there is no more issue. The issue would only occur if you hard code the import map in the webpack config.

> > - Performance problems may not manifest until the application is in an integrated environment.

Reviewer: What's an example of a performance problem that wouldn't occur in standalone mode, but would occur in an integrated environment?

Author: Just about any of them. Rerenders will be faster when the only JS on the page is the one application. Loading modules may be slower when there are others that might be blocking. My point here is just another variation of "local will be very different from prod".

Author: I ended up removing this line and mentioning it as one of the ways the differences between environments may manifest.

> > - All of these may add up leading to developer frustration and erosion of trust.

Reviewer: This doesn't seem to be a reason, but just a summary of the previous reasons. Here's a reason that might be helpful.

```suggestion
> - By default, this plugin defaults to not loading any global scripts, fonts, or css upfront in the HTML file. This can result in issues where the microfrontend looks or behaves differently in standalone mode versus when it is integrated.
```

Author: The sum of all the previous reasons can be its own reason, but I do like your point too. I'll add it as well.

> > - The core team reserves the right to deprecate this plugin based on the feedback we receive from single-spa users.

Reviewer: The language here was a bit stronger than I prefer ("reserve the right").

```suggestion
> - The single-spa core team may deprecate this plugin if it confuses more than helps.
```
